consul/src/main/java/com/linecorp/armeria/internal/consul/AgentCheckClient.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+/**
+ * Consul API client to registers, de-registers a check and gets list of check.
+ * {@code AgentCheckClient} is responsible for endpoint of Consul API:
+ * {@code `/agent/check`}(https://www.consul.io/api/agent/check.html)
+ */
+public final class AgentCheckClient {
```

Comment:
Question: This is only used in the test code. Did you make this for a testing purpose?
```suggestion
final class AgentCheckClient {
```

Reply:
Actually not. It is written for various use cases to check. But it is not necessary now. Would I better to remove it?

Reply:
I think it would be better to add it later if anyone requests the feature. 😀

Reply:
Ok!